src/pstl_offload.cpp

```diff
@@ -0,0 +1,137 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <new>
+#include <cassert>
+#include <cstdint>
+#include <sycl/sycl.hpp>
+
+#include <pstl_offload/internal/usm_memory_replacement_common.h>
+
+#if __linux__
+
+#define _PSTL_OFFLOAD_EXPORT __attribute__((visibility("default")))
+
+#include <dlfcn.h>
+#include <string.h>
+#include <unistd.h>
+
+namespace __pstl_offload {
+
+auto __get_original_free() {
+    using _free_type = void (*)(void*);
```

Comment:
Why do we have 1 leading underscore?

Reply:
I have just used the same approach for `using` as I saw in other oneDPL files:
Like [here](https://github.com/oneapi-src/oneDPL/blob/5aee5a571f98a206b4dd54adf6b6dfd9ca04a841/include/oneapi/dpl/pstl/parallel_backend_tbb.h#L734), [here](https://github.com/oneapi-src/oneDPL/blob/5aee5a571f98a206b4dd54adf6b6dfd9ca04a841/include/oneapi/dpl/pstl/glue_numeric_impl.h#L60) or [here](https://github.com/oneapi-src/oneDPL/blob/5aee5a571f98a206b4dd54adf6b6dfd9ca04a841/include/oneapi/dpl/pstl/memory_impl.h#L77)

Reply:
Changed as discussed